### DISCUSSION TITLE
How to add fullscreen button in Ionic 7 and React 18 app?

### DISCUSSION THREAD

**Discussion by norranorra** [vidstack/player#1285] [Q&A]
Hello, I am trying to display a video from youtube in my Ionic 7 and React 18 app but just cannot get the fullscreen button to work. I use custom layout and it work fine for play button. Are there any way to workaround it?

Here is my code and result using Custom Layout in iOS simulator (ignore the ugly style)

```ts
import { useRef } from 'react';
import { faPlay, faPause, faMinimize, faExpand } from '@fortawesome/pro-solid-svg-icons';
import { Controls, FullscreenButton, MediaPlayer, MediaPlayerInstance, MediaProvider, PlayButton, Poster, useMediaState } from '@vidstack/react';
import thumbnail from '@src/assets/images/graphics/video-thumbnail.svg';
import { Icon } from '../icon';

export const VideoPlayer = () => {
  const player = useRef<MediaPlayerInstance>(null);
  const isPaused = useMediaState('paused', player);
  const isFullscreen = useMediaState('fullscreen', player);

  return (
    <MediaPlayer
      ref={player}
      playsInline
      aspectRatio="16/9"
      src="https://www.youtube.com/embed/YN2H_sKcmGw"
      posterLoad="eager"
      className="media-player"
    >
      <MediaProvider>
        <Poster className="vds-poster" src={thumbnail} alt="movie trailer" />
      </MediaProvider>
      <Controls.Root className="vds-controls">
        <Controls.Group className="vds-controls-group" />
        <div className="vds-controls-spacer" />
        <Controls.Group className="vds-controls-group">
          <PlayButton className="vds-button">
            <Icon icon={isPaused ? faPlay : faPause} />
          </PlayButton>
        </Controls.Group>
        <div className="vds-controls-spacer" />
        <Controls.Group className="vds-controls-group">
          <FullscreenButton>
            <Icon icon={isFullscreen ? faMinimize : faExpand} />
          </FullscreenButton>
        </Controls.Group>
      </Controls.Root>
    </MediaPlayer>
  );
};

```
<img width="406" alt="Screenshot 2024-05-17 at 7 03 43 AM" src="https://github.com/vidstack/player/assets/89113489/d0eb21da-c4d0-46bf-939c-cea321f5ecf8">

**Comment by wsamoht** [comment]
I am implementing Vidstack in an Ionic Angular app using the default player. The fullscreen button doesn't appear on iOS. With a little research, it seems that the player is getting back from iOS that fullscreen is not supported.

```javascript
const { canFullscreen, fullscreen } = player.state;

try {
  await player.enterFullscreen()
} catch(e) {
  console.log(e)
}
```

`canFullscreen` returns `false`.

Trying to call `enterFullscreen()` returns "Error: [vidstack] fullscreen is not currently available on target `prefer-media`".

It looks like the Fullscreen API is not supported [here](https://caniuse.com/fullscreen) and [here](https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API#browser_compatibility) on iOS WebView. We are coming from Plyr which had a fallback of a "full window" which emulated a fullscreen. Vidstack must not have this fallback.

Looks like the only option on iOS is to set `playsinline=false`. That will then use the iOS native video player and force fullscreen on play.

**Issues:**
- https://github.com/vidstack/player/issues/1283
- https://github.com/vidstack/player/issues/1154

**Reply by wsamoht** [reply]
This does seem to contradict what is said [here](https://vidstack.io/docs/player/getting-started/installation/angular/?provider=video&styling=default-layout#11-playsinline-optional). Maybe something is different in the iOS WebView?

> The `playsinline` property will indicate that video content should be played inline (on mobile only), rather than launching the video in fullscreen mode on play. In addition, setting this property will also ensure custom player UI is displayed on iPhone when playing inline (hidden in fullscreen mode as native controls are forcefully displayed by the browser).

**Reply by wsamoht** [reply]
After a little more research, this issue only pertains to YouTube videos! Vimeo and normal MP4 videos work just fine when `playsinline=true` on iOS. Sounds like from [this](https://github.com/vidstack/player/issues/1154#issuecomment-1933245346) comment, YouTube doesn't have the API to send a fullscreen command.
